### ISSUE TITLE
YOLOV8 classification problem

### ISSUE THREAD

**Issue by 0zxq0** [ultralytics/ultralytics#22815]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am a beginner, and I am currently using YOLOv8 for classification tasks. I hope that the results of each class in the model I trained do not affect each other during detection, meaning that the probabilities are calculated separately, and their sum cannot equal one. Through searching for information, I have learned that I need to change the activation function from Softmax to sigmoid. However, I am unsure where to start. I hope someone can help and guide me on how to achieve this task

**Comment by UltralyticsAssistant** [comment]
👋 Hello @0zxq0, thank you for opening this issue and for your interest in Ultralytics 🚀! This is an automated response to help you get oriented quickly. An Ultralytics engineer will review your question and follow up with more specific guidance soon 🙂

Since your post is about a YOLO classification usage ❓ question (configuring activation / behavior for your classification task), it will be very helpful if you can share a bit more detail to assist the team when they arrive, for example:
- Your training command (Python or CLI)
- The model you are starting from (e.g., classification model variant)
- A brief description of your dataset and labels (single-label vs multi-label)
- A short code snippet showing how you run inference and interpret the outputs

In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that may help clarify how classification models are configured and how predictions are produced.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before debugging or changing configurations, please make sure you are on the latest `ultralytics` package (this often resolves unexpected behavior) and that your environment matches the recommended versions. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by 0zxq0** [comment]
I am using PyCharm and the model is yolov8l-cls.pt. My dataset is a single-label dataset, but I hope that the probabilities of each class do not affect each other during recognition
﻿

**Comment by verages** [comment]
> I am using PyCharm and the model is yolov8l-cls.pt. My dataset is a single-label dataset, but I hope that the probabilities of each class do not affect each other during recognition

<img width="1222" height="1036" alt="Image" src="https://github.com/user-attachments/assets/b7f748d7-8ed5-4a46-bd68-3879f11ede15" />

**Comment by verages** [comment]
<img width="1356" height="1190" alt="Image" src="https://github.com/user-attachments/assets/fe8e1bbb-ca13-4309-a826-948dc301df68" />

<img width="1284" height="584" alt="Image" src="https://github.com/user-attachments/assets/a802c52d-bc7d-4fe6-89b2-800d51ea4a4f" />

**Comment by 0zxq0** [comment]
> <img alt="图像" width="1356" height="1190" src="https://private-user-images.githubusercontent.com/87114157/519487047-fe8e1bbb-ca13-4309-a826-948dc301df68.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQyMTE4MDksIm5iZiI6MTc2NDIxMTUwOSwicGF0aCI6Ii84NzExNDE1Ny81MTk0ODcwNDctZmU4ZTFiYmItY2ExMy00MzA5LWE4MjYtOTQ4ZGMzMDFkZjY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI3VDAyNDUwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3NTY1MDZmMmFkNjkyNWJkM2E1MmFlYWEwMjA3MzkwODlkYzFlNDUyMmRmNGU4ZmZhMzkyNzFlMTM4YmQ1MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y0DNb7SFQFN_kRqkWR2vk5ppBjURSgeVzKVF1yzqGcs"> <img alt="图像" width="1284" height="584" src="https://private-user-images.githubusercontent.com/87114157/519487120-a802c52d-bc7d-4fe6-89b2-800d51ea4a4f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQyMTE4MDksIm5iZiI6MTc2NDIxMTUwOSwicGF0aCI6Ii84NzExNDE1Ny81MTk0ODcxMjAtYTgwMmM1MmQtYmM3ZC00ZmU2LTg5YjItODAwZDUxZWE0YTRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI3VDAyNDUwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2ZjMxNWU4MzU4N2ZhNDMzMjJiMmUxZmYzYmY2OWZmYTExOGNjMjMzZmUwNzlhMWQwYzM0YzZiM2FjYjY0YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.P1PM-KYUUTB6_7E3oAqKkM66Tc6cbozirAgZmBBP-zU">

Thank you for your guidance. I'll go implement it right away.

**Comment by glenn-jocher** [comment]
Sounds good! If you want more background while you implement, the differences and typical use cases of Sigmoid vs Softmax are summarized in the [Sigmoid function glossary](https://www.ultralytics.com/glossary/sigmoid) and the [Softmax activation function guide](https://www.ultralytics.com/glossary/softmax); if anything is unclear as you go, feel free to share a small code snippet here.
